app/src/components/navigation-bar.tsx

```diff
@@ -38,7 +39,12 @@ export function NavigationBar({
 }) {
   const { t } = useTranslation(lng, "navigation");
   const onChangeLanguage = (language: string) => {
-    i18next.changeLanguage(language);
+    Cookies.set("i18next", language);
+    const cookieLanguage = Cookies.get("i18next");
```

Comment:
Why are you reading the cookie immediately after writing it? This means it will always be `language`, so no need to read the cookie here imo.